packages/sdk/src/dataProtector/sharing/rentProtectedData.ts

```diff
@@ -1,56 +1,101 @@
 
 export const rentProtectedData = async ({
-  collectionTokenId,
+  iexec = throwIfMissing(),
+  graphQLClient = throwIfMissing(),
   protectedDataAddress,
-}: RentProtectedDataParams): Promise<SuccessWithTransactionHash> => {
+}: IExecConsumer &
+  SubgraphConsumer &
+  RentProtectedDataParams): Promise<SuccessWithTransactionHash> => {
   try {
-    const vCollectionId = positiveNumberSchema()
-      .required()
-      .label('collectionId')
-      .validateSync(collectionTokenId);
     const vProtectedDataAddress = addressOrEnsOrAnySchema()
       .required()
       .label('protectedDataAddress')
       .validateSync(protectedDataAddress);
+
+    const userAddress = (await iexec.wallet.getAddress()).toLowerCase();
+
+    const { protectedData, rentalParam } = await checkAndGetProtectedData({
+      graphQLClient,
+      protectedDataAddress: vProtectedDataAddress,
+      userAddress,
+    });
+
     const sharingContract = await getSharingContract();
-    let rentingParams;
-    try {
-      rentingParams = await sharingContract.protectedDataForRenting(
-        vCollectionId,
-        vProtectedDataAddress
-      );
-    } catch (error) {
-      throw new WorkflowError(
-        'Failed to get renting price from smart contract',
-        error
-      );
-    }
-    const price = rentingParams.price;
     const tx = await sharingContract.rentProtectedData(
-      vCollectionId,
-      vProtectedDataAddress,
+      protectedData.collection.id,
+      protectedData.id,
       {
-        value: price,
+        value: rentalParam.price,
         // TODO: See how we can remove this
         gasLimit: 900_000,
       }
     );
-    const txReceipt = await tx.wait();
+    await tx.wait();
     return {
       success: true,
-      txHash: txReceipt.hash,
+      txHash: tx.hash,
     };
   } catch (e) {
     throw new WorkflowError('Failed to rent Protected Data', e);
   }
 };
+
+async function checkAndGetProtectedData({
+  graphQLClient,
+  protectedDataAddress,
+  userAddress,
+}: {
+  graphQLClient: GraphQLClient;
+  protectedDataAddress: Address;
+  userAddress: Address;
+}) {
+  const { protectedData, rentalParam } = await getProtectedDataById({
+    graphQLClient,
+    protectedDataAddress,
+  });
+
+  if (!protectedData) {
+    throw new ErrorWithData(
+      'This protected data does not exist in the subgraph.',
+      { protectedDataAddress }
+    );
+  }
+
+  if (protectedData.owner.id !== DEFAULT_SHARING_CONTRACT_ADDRESS) {
+    throw new ErrorWithData(
+      'This protected data is not owned by the sharing contract.',
+      {
+        protectedDataAddress,
+        currentOwnerAddress: protectedData.owner.id,
+      }
+    );
+  }
+
+  if (protectedData.collection?.owner?.id !== userAddress) {
+    throw new ErrorWithData(
+      'This protected data is not part of a collection owned by the user.',
+      {
+        protectedDataAddress,
+        currentCollectionOwnerAddress: protectedData.collection?.owner?.id,
+      }
+    );
+  }
+
```

Comment:
Add a check about `isRentable`?

Reply:
8100bf988e3dcd7d5dff67f886f1deb51593c70e

Reply:
Some check like:

```
  if (!protectedData.isRentable) {
    throw new ErrorWithData('This protected data is not rentable.', {
      protectedDataAddress,
    });
  }
```

seems necessary to me.

Also in your added check, it looks like we get **all the rentals** for the protected data, not only the ones for the current user.
We probably need to filter by `rentals.renter === userAddress`.